parcels/tools/_helpers.py

```diff
@@ -152,3 +152,8 @@ def timedelta_to_float(dt: float | timedelta | np.timedelta64) -> float:
     if isinstance(dt, np.timedelta64):
         return float(dt / np.timedelta64(1, "s"))
     return float(dt)
+
+
+def calculate_next_ti(ti, tau, tdim):
+    """Check if the time is beyond the last time in the field"""
+    return tau > np.finfo(float).eps and ti < tdim - 1
```

Comment:
Could we rename to `should_calculate_next_ti` to be clearer?

And refactor to

```python
tau_is_significant = tau > np.finfo(float).eps
return tau_is_significant and ti < tdim - 1
```

Reply:
@erikvansebille  I was just thinking wouldn't `np.isequal(tau, 0.0)` have the save effect?

Reply:
Sure, but we would then need to do something like `tau_is_significant = not np.isequal(tau, 0.0)`; would that be more readable, with the `not`?

Reply:
I think `return not np.isequal(tau, 0) and ti<tdim-1` is good.

The reason I suggested before to refactor was since `tau > np.finfo(float).eps` wasn't immediately clear what was happening, but I think `np.isequal` is good